src/subcommand.rs

```diff
@@ -85,8 +85,19 @@ impl Subcommand {
 
     /// Reads gz mrt data from urls defined by range, decompresses them, parses mrt output, finds bottleneck.
     fn find_bottleneck(dump: &PathBuf, out: Option<&Path>) -> Result<()> {
-        let bottleneck = FindBottleneck::locate(dump)?;
-        bottleneck.write(out)?;
+        // Until the work is done, store results in this file so that
+        // a disrupted process doesn't look finalized.
+        let temp_result_file = tempfile();
+        let mut temp_result_file = match temp_result_file {
```

Comment:
Need to map this error instead of handling it this way. This is essentially no different than using `unwrap()` only with lots more code :p. 

I created a branch from this one and made a new error type. Looking at the [`tempfile` docs](https://docs.rs/tempfile/3.1.0/tempfile/fn.tempfile.html), we see that `tempfile()` returns a `Result<File>`, where the `Result` is a [`std::io::Result` type](https://doc.rust-lang.org/nightly/std/io/type.Result.html). This is great because we already know how to handle this. We could just use the `IoError` variant we have already defined in our custom `Error` enum, which would be fine, but I think it is nice to be more specific here and create another error variant just for failed temp file generation. 

I have cloned this branch, and when this is ready to merge, I am going to close this PR, make a new PR with all these changes you have made, plus the error handling, then merge it in. All the changes you have made will still be reflected properly.

Reply:
sounds good